tests/deployment_tests/gitlab.js

```diff
@@ -0,0 +1,201 @@
+const { gitlab } = require('../../deploy/configs/gitlab/gitlab.json');
+
+const adminUserName = gitlab.username;
+const adminPassword = gitlab.password;
+const gitlabHostname = gitlab.hostname;
+
+async function getToken(user = adminUserName, pass = adminPassword) {
+  let body;
+  try {
+    body = await request.post(`https://${gitlabHostname}/api/v4/session/`, { json: { login: user, password: pass } });
+  } catch (error) {
+    throw new Error('Error in obtaining a private token');
+  }
+  return body.private_token;
+}
+
+async function getId(userName) {
+  let userId = -1;
+  try {
+    const token = await getToken();
+    const body = await request.get(`https://${gitlabHostname}/api/v4/users?private_token=${token}`);
+    const entries = Array.from(JSON.parse(body));
+    for (let i = 0; i < entries.length; i += 1) {
+      if (entries[i].username === userName) {
+        userId = entries[i].id;
+      }
+    }
+  } catch (error) {
+    // this error will happen when the https call to gitlab fails
+    throw new Error(`Error in obtaining ID of user ${userName}`);
+  }
+  return userId;
+}
+
+async function listUsers() {
+  let usernames = [];
+  try {
+    const token = await getToken();
+    const body = await request.get(`https://${gitlabHostname}/api/v4/users?private_token=${token}`);
+    usernames = (JSON.parse(body)).map(entry => entry.username);
+  } catch (error) {
+    // this error will happen when the https call to gitlab fails
+    throw new Error('Error in listing users');
+  }
+  return usernames;
+}
+
+class User {
+  constructor(username, password) {
+    /* Default password if set to 12345678 */
+    this.username = username;
+    this.password = password || 12345678;
+  }
+
+  async addUser() {
+    const emailId = `${this.username}@autolab.com`;
+    let body = {};
+    try {
+      const token = await getToken();
+      body = await request.post(`https://${gitlabHostname}/api/v4/users?private_token=${token}`, {
+        json: {
+          name: this.username,
+          username: this.username,
+          password: this.password,
+          email: emailId,
+          skip_confirmation: true,
+        },
+      });
+    } catch (error) {
+      body = error.error;
+      if (error.message !== '409 - {"message":"Email has already been taken"}') {
+        throw new Error(`Error in adding user ${this.username}`);
+      }
+    }
+    return body;
+  }
+
+  async deleteUser() {
+    let body;
+    try {
+      const token = await getToken();
+      const userId = await getId(this.username);
+      body = await request.delete(`https://${gitlabHostname}/api/v4/users/${userId}?private_token=${token}`, { json: { id: userId, hard_delete: true } });
+    } catch (error) {
+      body = error.error;
+      if (error.message !== '404 - {"error":"404 Not Found"}') {
+        throw new Error(`Error in deleting user ${this.username}`);
+      }
+    }
+    return body;
+  }
+}
+
+class Project {
```

Comment:
Given how complex this class is, it makes sense to separate it into a dedicated module. Also, this class mixes user interaction with the model (actual work). Please see the review comments on [autolabcli PR-37](https://github.com/AutolabJS/autolabcli/pull/37) for possible ways of separating the UI from model and controller.

Given that gitlab.js is being written for deployment tests, it does not require user interaction at all. So we can just focus on the model in project.js and the way to use it in another module (may be gitlab/index.js).